File: yongkeun/4week/리스트자르기.js
function solution(n, slicer, num_list) {
  if (n === 1) {
    return num_list.slice(0, slicer[1] + 1); //0번 인덱스부터 b번 인덱스까지 추출하여 리턴
  } else if (n === 2) {
    return num_list.slice(slicer[0], num_list.length + 1);
  } else if (n === 3) {
    return num_list.slice(slicer[0], slicer[1] + 1);
  } else if (n === 4) {
    let answer = []; //새 배열을 만들고
    for (let i = slicer[0]; i < slicer[1] + 1; i += slicer[2]) {
      // slicer[2] 만큼의 간격을 줌
      answer.push(num_list[i]);
    }
    return answer;
  }
}

File: gang/week4/리스트 자르기.java
import java.util.ArrayList;

class Solution {
    public int[] solution(int n, int[] slicer, int[] num_list) {
        ArrayList<Integer> answer = new ArrayList<>();
        if(n == 1) {    // n이 1인 경우
            for(int i = 0; i<= slicer[1]; i++) {
                answer.add(num_list[i]);
            }
        }
        else if(n == 2) {   // n이 2인 경우
            for(int i = slicer[0]; i<num_list.length; i++) {
                answer.add(num_list[i]);
            }
        }
        else if(n == 3) {   // n이 3인 경우
            for(int i = slicer[0]; i<=slicer[1]; i++) {
                answer.add(num_list[i]);
            }
        }
        else {      // n이 4인 경우
            for(int i = slicer[0]; i<=slicer[1]; i += slicer[2]) {
                answer.add(num_list[i]);
            }
        }

        int[] result = new int[answer.size()];
        for(int i =0; i<answer.size(); i++) {   // ArrayList의 값을 다시 배열에 넣음
            result[i] = answer.get(i);
        }
        return result;
    }
}